### DISCUSSION TITLE
Commands in threads

### DISCUSSION THREAD

**Discussion by RobinLanglois** [Pycord-Development/pycord#1207] [Q&A] [answered]
Hello,

Is it possible to trigger a command (of `@commands.command`, not slash command) in a Discord thread ? Or is it absolutely necessary to use a slash command for that ?

Thank you in advance

**Comment by lorddusk** [comment]
Yes, you need to invite the bot into the thread however. 
Or make it force join any new threads:

@bot.event
async def on_thread_join(thread):
    await thread.join()

**Reply by RobinLanglois** [reply]
Well I did that and it doesn't seem to work.. is that a matter of version maybe ?

**Reply by lorddusk** [reply]
If you have the latest beta, you also need to make sure your bot has the message_contents intent enabled.

**Reply by RobinLanglois** [reply]
I'm in 1.7.3
I think it does but I'm going to check that

**Reply by lorddusk** [reply]
Ah. 1.7.3 has no thread support.
You need the beta.

**Reply by RobinLanglois** [reply]
Yes it does have the intent

**Reply by RobinLanglois** [reply]
oh ok, is there a way to install it via pip ?

**Reply by lorddusk** [reply]
pip install py-cord==2.0.0b5

**Reply by RobinLanglois** [reply]
I tried and this still doesn't work :confused:

**Reply by BobDotCom** [reply]
The bot should automatically join the thread, does it have access to join?

**Reply by RobinLanglois** [reply]
@BobDotCom sorry for delay. The bot is in the thread but it doesn't answer to any command in a thread

**Comment by Lulalaby** [comment, marked as answer]
Best would be you join https://discord.gg/pycord
